src/diffusers/pipelines/consistency_models/pipeline_consistency_models.py

```diff
@@ -0,0 +1,317 @@
+            True, otherwise a `tuple. When returning a tuple, the first element is a list with the generated images.
+        """
+        # 0. Prepare call parameters
+        img_size = self.unet.config.sample_size
+        device = self._execution_device
+
+        # 1. Check inputs
+        self.check_inputs(num_inference_steps, timesteps, latents, batch_size, img_size, callback_steps)
+
+        # 2. Prepare image latents
+        # Sample image latents x_0 ~ N(0, sigma_0^2 * I)
+        sample = self.prepare_latents(
+            batch_size=batch_size,
+            num_channels=3,
+            height=img_size,
+            width=img_size,
+            dtype=self.unet.dtype,
+            device=device,
+            generator=generator,
+            latents=latents,
+        )
+
+        # 3. Handle class_labels for class-conditional models
+        class_labels = self.prepare_class_labels(batch_size, device, class_labels=class_labels)
+
+        # 4. Prepare timesteps
+        if timesteps is not None:
+            self.scheduler.set_timesteps(timesteps=timesteps, device=device)
+            timesteps = self.scheduler.timesteps
+            num_inference_steps = len(timesteps)
+        else:
+            self.scheduler.set_timesteps(num_inference_steps)
+            timesteps = self.scheduler.timesteps
+
+        # 5. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+        extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+
+        # 6. Denoising loop
+        # Multistep sampling: implements Algorithm 1 in the paper
+        # Onestep sampling does not use random noise
+        use_noise = False if num_inference_steps == 1 else True
+        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+        with self.progress_bar(total=num_inference_steps) as progress_bar:
+            for i, t in enumerate(timesteps):
+                scaled_sample = self.scheduler.scale_model_input(sample, t)
+                scaled_t = self.scheduler.scale_timestep(t)
+                model_output = self.unet(scaled_sample, scaled_t, class_labels=class_labels).sample
+
+                sample = self.scheduler.step(
+                    model_output, t, sample, use_noise=use_noise, **extra_step_kwargs
+                ).prev_sample
+
+                # call the callback, if provided
+                if i == len(timesteps) - 1 or ((i + 1) > num_warmup_steps and (i + 1) % self.scheduler.order == 0):
+                    progress_bar.update()
+                    if callback is not None and i % callback_steps == 0:
+                        callback(i, t, sample)
+
+        # 7. Post-process image sample
+        sample = (sample / 2 + 0.5).clamp(0, 1)
+        sample = sample.cpu().permute(0, 2, 3, 1).numpy()
+
+        if output_type == "pil":
+            sample = self.numpy_to_pil(sample)
+
```

Comment:
```suggestion
        sample = (sample / 2 + 0.5).clamp(0, 1)
        sample = sample.cpu().permute(0, 2, 3, 1).numpy()

        if output_type == "pil":
            sample = self.numpy_to_pil(sample)

```

Can we also allow to return `"pt"` tensors here as well as `"np"`?

Reply:
I have refactored the code to move the postprocessing logic into a `postprocess_image` pipeline method.

I think there might be some opportunity for code reuse here - `postprocess_image` heavily follows the logic from [`VaeImageProcessor.postprocess`](https://github.com/huggingface/diffusers/blob/main/src/diffusers/image_processor.py#L215), and the logic seems generally applicable. Not sure what the right design should be, but perhaps a method similar to `postprocess_image` could be added to `DiffusionPipeline` (similar to `DiffusionPipeline.numpy_to_pil`) or a class like `VaeImageProcessor` could be used for general image generation models.